examples/ecs/system_piping.rs

```diff
@@ -19,7 +19,8 @@ fn main() {
         .add_systems(
             Update,
             (
-                parse_message_system.pipe(handler_system),
+                parse_message_system
+                    .pipe::<_, In<Result<usize, ParseIntError>>, _, _>(handler_system),
```

Comment:
I've been wanting this flexibility for awhile and I like that this opens doors, but I find this regression pretty concerning.

Reply:
Very cool effort nonetheless. Well done!

Reply:
Type inference should now be restored to its former glory in the latest set of changes! 😄